parlai/core/torch_agent.py

```diff
@@ -137,10 +131,50 @@ def __init__(
             candidates=candidates,
             candidate_vecs=candidate_vecs,
             image=image,
-            observations=observations,
+            _context_original_length=_context_original_length,
+            _context_truncate_rate=_context_truncate_rate,
+            _label_original_length=_label_original_length,
+            _label_truncate_rate=_label_truncate_rate,
             **kwargs,
         )
 
+    def to(self, dev):
+        """
+        Move all tensors in the batch to a device.
+
+        Happens in place. Note that valid_indices and fields starting with an
```

Comment:
i like these semantics. can we make that more clear in the Batch object description? specifically underscored fields being exempt